python/lsst/analysis/drp/pipelineBuilder.py

```diff
@@ -0,0 +1,151 @@
+# This file is part of analysis_drp.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from textwrap import dedent
+from typing import Iterable, Mapping, Optional, Sequence
+
+from lsst.resources import ResourcePath
+from lsst.pipe.base.pipelineIR import PipelineIR, LabeledSubset
+
+
+class PlotPipelineBuilder:
+    """Class to support building a complete analysis drp pipeline from an input
+    file.
+    """
+
+    def __init__(self):
+        # Setup the command line argument parser that will be used to build
+        # the pipeline
+        self.parser = argparse.ArgumentParser()
+        self.parser.add_argument("pipeline", help="input pipeline to process")
+
+    def getGroups(self) -> Mapping[str, tuple[Optional[str], Iterable[str]]]:
```

Comment:
If this is about defining subsets, maybe just call it "subsets" rather than "groups"?